spec/unit/type/user_spec.rb

```diff
@@ -325,24 +325,31 @@ def self.instances; []; end
   end
 
   describe "when managing passwords" do
-    before do
-      @password = described_class.new(:name => 'foo', :password => 'mypass').parameter(:password)
-    end
+    let(:transaction) { Puppet::Transaction.new(Puppet::Resource::Catalog.new, nil, nil) }
+    let(:harness) { Puppet::Transaction::ResourceHarness.new(transaction) }
+    let(:resource) { described_class.new(:name => 'foo', :password => 'mypass') }
 
     it "should not include the password in the change log when adding the password" do
-      expect(@password.change_to_s(:absent, "mypass")).not_to be_include("mypass")
+      status = harness.evaluate(resource)
+      sync_event = status.events[0]
+      puts status.events
```

Comment:
puts

Reply:
these aren't passing at all yet :( I only opened this PR to get help from @jhelwig :-D

Reply:
Ah. Sorry :)

Reply:
No worries! If you want to help too, that'd be awesome!